packages/opentelemetry-propagation-utils/src/pubsub-propagation.ts

```diff
@@ -50,15 +50,26 @@ const patchArrayFilter = (
     return newArray;
   };
 
-  Object.defineProperty(messages, 'filter', {
+  Object.defineProperty(messages, "filter", {
     enumerable: false,
     value: patchedFunc,
   });
 };
 
+function isPromise(value: unknown): value is Promise<unknown> {
+  return value != null && typeof (value as any).then == "function";
```

Comment:
We prefer to use `===` instead of `==`
optional nit: I think it can be shortened to `typeof (value as any)?.then === 'function'`.